devops/scripts/benchmarks/benches/gromacs.py

```diff
@@ -0,0 +1,335 @@
+# See LICENSE.TXT
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import os
+import subprocess
+import tarfile
+import urllib.request
+from pathlib import Path
+from .base import Suite, Benchmark
+from options import options
+from utils.utils import git_clone
+from utils.result import Result
+
+
+class GromacsBench(Suite):
+    GROMACS_REPO = "https://gitlab.com/gromacs/gromacs.git"
+    GROMACS_TAG = "v2025.1"
+    GRAPPA_BENCHMARKS_URL = (
+        "https://zenodo.org/record/11234002/files/grappa-1.5k-6.1M_rc0.9.tar.gz"
+    )
+
+    def __init__(self, directory):
+        # Initialize GromacsBench-specific attributes
+        self.directory = Path(directory).resolve()
+        self.gromacs_dir = self.directory / "gromacs"
+        self.grappa_dir = self.directory / "grappa-1.5k-6.1M_rc0.9"
+        self.build_dir = self.gromacs_dir / "build"
+
+    def name(self):
+        return "Gromacs Bench"
+
+    def benchmarks(self) -> list[Benchmark]:
+        systems = [
+            "0001.5",
+            "0003",
+            "0006",
+            "0012",
+            "0024",
+            "0048",
+            "0096",
+            "0192",
+            "0384",
+        ]
+        return [
+            GromacsSystemBenchmark(self, system, self.gromacs_dir, self.grappa_dir)
+            for system in systems
+        ]
+
+    def setup(self):
+        print(f"Working directory: {self.directory}")
+        self.directory.mkdir(parents=True, exist_ok=True)
+
+        if not self.gromacs_dir.exists():
+            print(
+                f"Cloning GROMACS repository (tag: {self.GROMACS_TAG}) into {self.gromacs_dir}..."
+            )
+            repo_path = git_clone(
+                self.directory,
+                "gromacs-repo",
+                self.GROMACS_REPO,
+                self.GROMACS_TAG,
+            )
+            print(f"GROMACS repository cloned to {repo_path}")
+        else:
+            print(f"GROMACS repository already exists at {self.gromacs_dir}")
+
+        # Build GROMACS
+        self.build_dir.mkdir(parents=True, exist_ok=True)
+        print(f"Building GROMACS in {self.build_dir}...")
+        subprocess.run(
+            [
+                "cmake",
+                "../",
+                f"-DCMAKE_BUILD_TYPE=Release",
+                f"-DCMAKE_CXX_COMPILER={options.sycl}/bin/clang++",
+                f"-DCMAKE_C_COMPILER={options.sycl}/bin/clang",
+                f"-DGMX_GPU=SYCL",
+                f"-DGMX_SYCL_ENABLE_GRAPHS=ON",
+                f"-DGMX_FFT_LIBRARY=MKL",
+                f"-DGMX_BUILD_OWN_FFTW=ON",
+                f"-DGMX_GPU_FFT_LIBRARY=MKL",
+                f"-DGMX_GPU_NB_CLUSTER_SIZE=8",
+                f"-DGMX_OPENMP=OFF",
+            ],
+            check=True,
+            cwd=self.build_dir,  # Ensure the command runs in the build directory
+        )
+        subprocess.run(["make", "-j"], check=True, cwd=self.build_dir)
```

Comment:
`cmake --build {self.build_dir} -j {options.build_jobs}` for consistency with other benchmarks.